apps/examples/testcase/le_tc/network/tc_net_setsockopt.c

```diff
@@ -647,22 +516,91 @@ static void tc_net_setsockopt_invalid_level_n(int s)
 * Preconditions         :socket file descriptor
 * Postconditions        :
 * @return               :void
-
+*/
 static void tc_net_setsockopt_invalid_opt_name_n(int s)
 {
-    int ret = -1;
-    //int optval=1;
+	int ret = -1;
+
+	ret = setsockopt(s, SOL_SOCKET, TCP_NODELAY, 0, 0);
+	TC_ASSERT_EQ("setsockopt", ret, -1);
+	TC_SUCCESS_RESULT();
+}
+
+static void tc_net_setsockopt_solsocket_p(int fd)
+{
+	tc_net_setsockopt_keepalive_p(fd);
+	tc_net_setsockopt_broadcast_p(fd);
+#if CONFIG_NET_SO_RCVBUF
+	tc_net_setsockopt_rcvbuf_p(fd);
+#endif
+#if CONFIG_NET_SO_REUSE
+	tc_net_setsockopt_reuseaddr_p(fd);
+#endif
+#if CONFIG_NET_SO_RCVTIMEO
+	tc_net_setsockopt_rcvtimo_p(fd);
+#endif
+#if CONFIG_NET_SO_SNDTIMEO
+	tc_net_setsockopt_sndtimo_p(fd);
+#endif
+#if CONFIG_NET_UDP
+	tc_net_setsockopt_no_check_p();
+#endif
+}
+
+#if CONFIG_NET_LWIP_IGMP && CONFIG_NET_UDP
+static void tc_net_setsockopt_multicast_p(void)
+{
+	char *ip = "127.0.0.1";
+	int udp_fd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (udp_fd < 0) {
+		printf("socket creation error (%s) line:%d\n", __FUNCTION__, __LINE__);
+		return;
+	}
+	tc_net_setsockopt_multicast_add_group_p(udp_fd);
+	tc_net_setsockopt_multicast_drop_group_p(udp_fd);
+
+	tc_net_setsockopt_multicast_if_p(udp_fd, ip);
+	tc_net_setsockopt_multicast_ttl_loop_p(udp_fd);
+	tc_net_setsockopt_multicast_ttl_loop_own_p(udp_fd);
+	tc_net_setsockopt_multicast_ttl_p(udp_fd);
+	close(udp_fd);
+}
+#endif
 
-    ret = setsockopt(s, SOL_SOCKET, SCTP_AUTOCLOSE, 0, 0);
-    if (ret != -1) {
-		printf("tc_net_setsockopt_invalid_opt_name_n FAIL: setopt KEEPALIVE failure\n");
-		nw_total_fail++;
-		RETURN_ERR;
-    }
+static void tc_net_setsockopt_ip_p(int fd)
+{
+	tc_net_setsockopt_ip_ttl_p(fd);
+	tc_net_setsockopt_ip_tos_p(fd);
+#if CONFIG_NET_LWIP_IGMP && CONFIG_NET_UDP
+	tc_net_setsockopt_multicast_p();
+#endif
+}
 
-    printf("tc_net_setsockopt_invalid_opt_name_n PASS\n");
-    nw_total_pass++;
-}*/
+static void tc_net_setsockopt_tcp_p(int fd)
+{
+#if CONFIG_NET_TCP_KEEPALIVE
+	tc_net_setsockopt_multicast_tcp_keepcnt_p(fd);
+	tc_net_setsockopt_multicast_tcp_keepintvl_p(fd);
+	tc_net_setsockopt_multicast_tcp_keepidle_p(fd);
+#endif
+	tc_net_setsockopt_multicast_tcp_keepalive_p(fd);
+	tc_net_setsockopt_multicast_tcp_nodelay_p(fd);
+}
+
+static void tc_net_setsockopt_n(int fd)
```

Comment:
Every tc includes TC_ASSERT_XX macro. These functions are named "tc_xxxx" but they don't include the macro. Then, you should name it as "_main".

Reply:
That's the naming rule we should follow in testcase.